### ISSUE TITLE
allow custom bwa mem options (e.g. "-x pacbio") during long-read scaffolding stage

### ISSUE THREAD

**Issue by benvvalk** [bcgsc/abyss#83]
For next ABySS release

**Comment by sjackman** [comment]
What should the option be called? `BWAMEM_OPTIONS`?

**Comment by benvvalk** [comment]
Sure.  Or maybe `long_mapopt`.

**Comment by benvvalk** [comment]
I'll try to do something that matches the existing variable-naming conventions.

**Comment by sjackman** [comment]
Caps is preferred. Lower case is for internal variables. How about `LONG_BWAMEM_OPTIONS`?
I rather wish that I had picked something shorter than `…_OPTIONS`, but for consistency's sake.

**Comment by benvvalk** [comment]
Yep, I realized after responding that user options should be in caps.  `LONG_BWAMEM_OPTIONS` sounds good to me.

**Comment by macmanes** [comment]
The only thing I'd add is to ask about the user that has nanopore, a transcriptome and a PacBio dataset to use.  I guess you could have users enter them as a comma/space delimited list, e.g., `long='nanopore.fasta transcriptome.fasta PacBio.fasta'` and `LONG_BWAMEM_OPTIONS='ont2d default pacbio'` which would pass the appropriate things to `-x`.

**Comment by sjackman** [comment]
The `long` option should have taken a list of library names rather than a list of file names, to be consistent with the other options, like `pe` and `mp`. I think we can still make that change. Then you can have
`long='pacbio nanopore' pacbio_LONG_BWAMEM_OPTIONS=-xpacbio nanopore_ LONG_BWAMEM_OPTIONS=-xnanopore`
These latter two could be the default values for these variables.

**Comment by sjackman** [comment]
I'd still like to fix this issue.

**Comment by sjackman** [comment]
Shh stalebot
